src/adhocracy_core/adhocracy_core/auditing/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""Auditlog of events stored in a ZODB database."""
+import transaction
+import json
+import substanced.util
+
+from pyramid.traversal import resource_path
+from BTrees.OOBTree import OOBTree
+from datetime import datetime
+from logging import getLogger
+from collections import namedtuple
+from adhocracy_core.utils import get_user
+from adhocracy_core.sheets.principal import IUserBasic
+from adhocracy_core.utils import get_sheet_field
+
+
+logger = getLogger(__name__)
+
+AuditEntry = namedtuple('AuditEntry', ['name', 'payload'])
+
+RESOURCE_CREATED = 'resourceCreated'
+RESOURCE_MODIFIED = 'resourceModified'
+
+
+class AuditLog(OOBTree):
+
+    """An Auditlog composed of entries."""
+
+    def add(self, name, **kw):
+        """ Add a record the audit log.
+
+        ``_name`` should be the event name,
+        ``_oid`` should be an object oid or ``None``, and ``kw`` should be a
+        json-serializable dictionary.
+        """
```

Comment:
methods shoud have type annotations (similar syntax like  mypy) for documentation and static type checking.  
If the method arguments are not self explaining or the functions raises Exception you can add additional information with javadoc-style in the docstring.

Why does this function accept arbitrary keywords?